### ISSUE TITLE
Dash in PowerShellGallery project name cause invalid response data

### ISSUE THREAD

**Issue by mkellerman** [badges/shields#2929]
Are you experiencing an issue with...

- [X] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: **Description**

Seems like having a '-' dash in the project name in powershellgallery prevents it to show the badge correctly.

:link: **Link to the badge**

https://img.shields.io/powershellgallery/v/Invoke-CommandAs.svg

**Comment by paulmelnikow** [comment]
Hi, thanks for the report. 

Looks like the problem is not the dash, but that Version is a number instead of a string. Should we use NormalizedVersion instead? 

```
 Service class  👩‍🍳
 PowershellGalleryVersion
 Named params  🎫
 { which: 'v', packageName: 'Invoke-CommandAs' }
 Query params  🖍
 { colorA: undefined,
  colorB: undefined,
  label: undefined,
  link: undefined,
  logo: undefined,
  logoColor: undefined,
  logoPosition: undefined,
  logoWidth: undefined,
  style: undefined }
 Request  🏹
 https://www.powershellgallery.com/api/v2/Search()
 Response status code  🎯
 200
 Response XML (before validation)  🎯
{ feed:
   { id: 'http://schemas.datacontract.org/2004/07/',
     title: '',
     updated: '2019-02-05T16:09:39Z',
     link: '',
     entry:
      { id:
         'https://www.powershellgallery.com/api/v2/Packages(Id=\'Invoke-CommandAs\',Version=\'3.0.0\')',
        category: '',
        link: [ '', '' ],
        title: 'Invoke-CommandAs',
        updated: '2019-02-05T16:00:23Z',
        author: { name: 'Marc R Kellerman' },
        content: '',
        'm:properties':
         { 'd:Id': 'Invoke-CommandAs',
           'd:Version': 3,
           'd:NormalizedVersion': '3.0.0',
           'd:Authors': 'Marc R Kellerman',
           'd:Copyright': '(c) 2017 MKellerman. All rights reserved.',
           'd:Created': '2019-02-05T06:19:56.99',
           'd:Dependencies': '',
           'd:Description':
            'Invoke Command as System/User on Local/Remote computer using ScheduleTask.',
           'd:DownloadCount': 822,
           'd:GalleryDetailsUrl':
            'https://www.powershellgallery.com/packages/Invoke-CommandAs/3.0.0',
           'd:IconUrl': '',
           'd:IsLatestVersion': true,
           'd:IsAbsoluteLatestVersion': true,
           'd:IsPrerelease': false,
           'd:Language': '',
           'd:LastUpdated': '2019-02-05T16:00:23.547',
           'd:Published': '2019-02-05T06:19:56.99',
           'd:PackageHash':
            'kTdNRQwa54Q7RWzwK+irggzMj/s6RhT9NvMGDAh0f62PGBrIImqJYFxAa8vFOnsHanEGYeerBmvZLMxabOcamg==',
           'd:PackageHashAlgorithm': 'SHA512',
           'd:PackageSize': 10192,
           'd:ProjectUrl': '',
           'd:ReportAbuseUrl':
            'https://www.powershellgallery.com/packages/Invoke-CommandAs/3.0.0/ReportAbuse',
           'd:ReleaseNotes': '',
           'd:RequireLicenseAcceptance': false,
           'd:Summary': '',
           'd:Tags':
            'PSModule PSFunction_Invoke-CommandAs PSCommand_Invoke-CommandAs PSIncludes_Function',
           'd:Title': '',
           'd:VersionDownloadCount': 6,
           'd:MinClientVersion': '',
           'd:LastEdited': '',
           'd:LicenseUrl': '',
           'd:LicenseNames': '',
           'd:LicenseReportUrl': '',
           'd:ItemType': 'Module',
           'd:FileList':
            'Invoke-CommandAs.nuspec|Invoke-CommandAs.psd1|Invoke-CommandAs.psm1|Private\\Invoke-ScheduledTask.ps1|Public\\Invoke-CommandAs.ps1',
           'd:GUID': '9b7281cf-c80f-44bb-96c0-ed1137056164',
           'd:PowerShellVersion': '',
           'd:PowerShellHostVersion': '',
           'd:DotNetFrameworkVersion': '',
           'd:CLRVersion': '',
           'd:ProcessorArchitecture': '',
           'd:CompanyName': 'Marc R Kellerman',
           'd:Owners': 'mkellerman' } } } }
 Response did not match schema  🤷
 child "feed" fails because [child "entry" fails because [child "m:properties" fails because [child "d:Version" fails because ["d:Version" must be a string]]]]
 Handled error  🙅
 { ValidationError: child "feed" fails because [child "entry" fails because [child "m:properties" fails because [child "d:Version" fails because ["d:Version" must be a string]]]]
    at Object.exports.process (/Users/pnm/code/shields/node_modules/joi/lib/errors.js:203:19)
    at internals.Object._validateWithOptions (/Users/pnm/code/shields/node_modules/joi/lib/types/any/index.js:764:31)
    at module.exports.internals.Any.root.validate (/Users/pnm/code/shields/node_modules/joi/lib/index.js:147:23)
    at validate (/Users/pnm/code/shields/core/base-service/validate.js:28:32)
    at Function._validate (/Users/pnm/code/shields/core/base-service/base.js:332:12)
    at PowershellGalleryVersion._requestXml (/Users/pnm/code/shields/core/base-service/base-xml.js:39:29)
    at process._tickCallback (internal/process/next_tick.js:68:7) prettyMessage: 'invalid response data' }
 Service data  🛡
 { isError: true,
  message: 'invalid response data',
  color: 'lightgray' }
```

**Comment by mkellerman** [comment]
If you're asking me, I have no idea..But I'm guessing it should always be treated as a string, since you might have versions with 2.0.0.1-alpha and such in the gallery.
